### DISCUSSION TITLE
How to retain files between Plan and Apply

### DISCUSSION THREAD

**Discussion by ValeriiVozniuk** [terrakube-io/terrakube#1120] [Q&A] [answered]
We are testing an option to install additional applications via modules depending on workspace configuration. One of them is LinkerD, which needs TLS certificates to be generated before deployment. And here lies the problem: if I run generate script on Plan step, they are absent on Apply and Terraform is failing. And if I try to generate them on apply, the Plan is failing due to missing files. And if I run it on both steps, Apply if failing due to values mismatch between Plan and Apply generated files. And I can't run just Apply because it complains about missing plan file.
So the question is: how to retain files created on Plan step to use them on Apply step? :) Move out of workspace and copy back, or there is some more elegant solution?
PS: it would be nice to have terraformPlanAndApply FlowType [here](https://github.com/AzBuilder/terrakube/blob/main/api/src/main/java/org/terrakube/api/plugin/scheduler/job/tcl/model/FlowType.java), so both operations could be run in a single step.

**Comment by alfespa17** [comment, marked as answer]
The option right now is to save those file like this :

https://github.com/AzBuilder/terrakube/issues/852

**Reply by ValeriiVozniuk** [reply]
I see, thank you :)
